client/src/modules/list/components/Items/ListItemDescription/ListItemDescription.jsx

```diff
@@ -0,0 +1,285 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import _trim from 'lodash/trim';
+import _isEqual from 'lodash/isEqual';
+import Linkify from 'react-linkify';
+import Textarea from 'react-textarea-autosize';
+import classNames from 'classnames';
+
+import Preloader, {
+  PreloaderSize,
+  PreloaderTheme
+} from 'common/components/Preloader';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+import { updateListItem } from '../model/actions';
+import { KeyCodes } from 'common/constants/enums';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+
+class ListItemDescription extends PureComponent {
+  pendingPromise = null;
+
+  constructor(props) {
+    super(props);
+
+    const { description } = this.props;
+    const trimmedDescription = description.trim();
+
+    this.state = {
+      isDescriptionUpdated: false,
+      descriptionInputValue: trimmedDescription,
+      isDescriptionEdited: false,
+      isFocused: false,
+      pending: false
+    };
+
+    this.descriptionTextarea = React.createRef();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    this.checkIfDescriptionUpdated();
+    const { isDescriptionEdited: previousIsDescriptionEdited } = prevState;
+    const { isDescriptionEdited } = this.state;
+
+    if (!previousIsDescriptionEdited && isDescriptionEdited) {
+      this.descriptionTextarea.current.focus();
+      this.handleFocus();
+    }
+
+    if (previousIsDescriptionEdited && !isDescriptionEdited) {
+      this.handleBlur();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  handleKeyPress = event => {
+    const { code } = event;
+    const { isFocused } = this.state;
+
+    if (isFocused && code === KeyCodes.ESCAPE) {
+      this.handleBlur();
+    }
+  };
+
+  handleFocus = () => {
+    this.setState({ isFocused: true });
+    document.addEventListener('keydown', this.handleKeyPress);
+  };
+
+  handleBlur = () => {
+    this.setState({ isFocused: false });
+    document.removeEventListener('keydown', this.handleKeyPress);
+    this.handleDescriptionUpdate();
+  };
+
+  handleStartEditing = () => {
+    this.setState({ isDescriptionEdited: true });
+  };
```

Comment:
do you need this method and property?
can't you assume that if an input has the focus it is also edited
and if it has blur it is not edited

Reply:
Yes, it's necessary. But textarea isn't always displayed. When there is no description add-button is displayed or if there is description we display a paragraph with a description to replace entered link by an anchor tag (we can't to that inside input/textarea). So when user clicks add-button or description div, this handler is invoked set `isDescriptionEdited: true` and render knows that textarea has to be displayed.